code/entities/machine/soda-fountain/SodaFountain.CupFillable.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class CupFillable : ModelEntity
+{
+    public enum MaterialGroup : int
+    {
+        Default = 0,
+        Blue = 1,
+        Green = 2,
+        Red = 3,
+        Black = 4
+    }
+
+    static public string CupItemUniqueId = "soda"; 
```

Comment:
You're doing this beautiful code down in `Dispenser.cs`
```
private string SodaDispenseParticlePath => SodaType switch
    {
        SodaFountain.SodaType.Conk => "particles/soda_fountain/walker/sodafill1_nocup_conk_f.vpcf",
        SodaFountain.SodaType.MionPisz => "particles/soda_fountain/walker/sodafill1_nocup_mionpisz_f.vpcf",
        SodaFountain.SodaType.Spooge => "particles/soda_fountain/walker/sodafill1_nocup_spooge_f.vpcf",

        _ => "particles/soda_fountain/walker/sodafill1_nocup_conk_f.vpcf"
    };
```

And then you declare 4 strings here :(

Do the same thing you did down there, create a mapping from `SodaType` => `string` if you're going to do something like this. It also gets rid of the unnecessary switch code in `Platform.cs` ~line 94

Reply:
Moved this into a static function in the SodaFountain class so you are able to retrieve the particle path based on the soda type